tests/test_which.py

```diff
@@ -1,6 +1,7 @@
 
+import sys
 from jupyter_packaging.setupbase import which
 
 
-def test_which_finds_python():
-    assert which('python')
+def test_which_finds_python_executable():
+    assert which(sys.executable)
```

Comment:
From the [python docs](https://docs.python.org/3/library/sys.html#sys.executable) (emphasis added):
> sys.executable
> 
>    A string giving the ### absolute path of the executable binary for the Python interpreter, [...]

If `sys.executable` is an absolute path, I'm not sure if we are presenting `which` with a very though challenge? :)

Reply:
oops... :) so what about

split_path = sys.executable.split('/')
assert which(split_path[-1])

The problem is that 'python' indeed doesn't have to exist on some systems which run only on Python 3 (since Python 2 is officially EOL now)

Reply:
I would use `os.path.basename` as that will handle things the cleanest, but yes that would probably be a better test case (on Windows that would typically resolve to `python.exe`, but that should still work).